doc/metadata-server/README.md

```diff
@@ -0,0 +1,327 @@
+
+### `transaction_metadatum`
+
+`transaction_metadatatum` can be nearly any `CBOR` encoded data (with string / byte buffers length limitations to `64B`). [Tx metadata specification](https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/tx-metadata.md#metadata-schemas---mappings-and-formats) defines a way to encode metadata using `cardano-cli` and `JSON` encoded payload. For the purpose of this draft we use simplified ["no schema"](https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/tx-metadata.md#no-schema) style spec (just plain `JSON`) of the format which is more readble and can be quickly tested with `cardano-cli`.
+We should expand this in the next iterations into a [detailed schema](https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/tx-metadata.md#detailed-schema) which is translated into a precise CBOR representation (both `cardano-cli` and [`node-wallet`](https://input-output-hk.github.io/cardano-wallet/user-guide/common-use-cases/TxMetadata) support it). We want to use precise CBOR encoding because this exchange protocol should be tool and language agnostic.
+
+#### Storage
+
+Using blockchain as the only source of data has many advantages - we don't have to rely on things like hosting, domain registration, maintenance of the data storage systems etc. On the other hand, storing data on-chain has its cost because it increases transaction fees. `TxMetadata` size contributes to the overall transaction size (without any additional fees). Additionally transactions have currently [64KB](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0009/README.md#updatable-protocol-parameters) size limit which is an additional "risky" factor.
+
+We try to support both approaches of metadata distribution i.e. on-chain and off-chain by using optional keys in the schema. Of course on-chain metadata are immutable and we don't provide update mechanismfor them (i.e. [NFT metadata in theory supports updates](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0025#update-metadata-link-for-a-specific-token)). Instead we allow usage of both versions and off-chain metadata source (if responsive) can be used for metadata updates.
+
+### Draft of the metadata schema
+
+Following [CIP-0026 convention](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0026):
+
+- We use [_Blake2b-256_](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0026#hash-function),
+
+- We use [this encoding](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0026#attestation-signatures) to construct messages for signing.
```

Comment:
Could we generalize this to multiple signature and hash methods? Perhaps there could be a "type" field that specifies which CIP the signature conforms to, or whether it is a W3C DID?